nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -455,6 +455,7 @@ ObstacleLayer::updateFootprint(
   double * max_x,
   double * max_y)
 {
+  std::lock_guard<Costmap2D::mutex_t> guard(*getMutex());
```

Comment:
Why is this costmap layer not having any dynamic parameters? I see the voxel does, which derives from this, and would implicitly add those dynamic parameters to both sets of layers at once

Reply:
Yes, so we originally started off the PR with Obstacle Layer out of scope of these changes (ref Alexey's comments above) but then we realised that it wasnt possible to make voxel dynamic without also making obstacle layer dynamic.

But then we decided that since this PR is already pretty big, reviewing it and keeping it in sync with master is becoming a problem so we think its best if we target ObstacleLayer in the next PR. Is this okay?

Reply:
OK!